packages/grafana-ui/src/components/DataSourceSettings/DataSourceHttpSettings.tsx

```diff
@@ -308,6 +310,7 @@ export const DataSourceHttpSettings = (props: HttpSettingsProps) => {
           <CustomHeadersSettings dataSourceConfig={dataSourceConfig} onChange={onChange} />
         )}
       </>
+      {secureSocksDSProxyEnabled && <SecureSocksProxySettings options={dataSourceConfig} onOptionsChange={onChange} />}
```

Comment:
In regards to external plugins there's possibly a problem I see with this approach, but maybe external plugins is not a concern for now? If a plugin author would provide `secureSocksDSProxyEnabled=true` to the DataSourceHttpSettings component it would always render even if the secure socks configuration is not enabled in Grafana config, right?